cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
```

Comment:
> Also, I am confused why you remove the different build types here.

Because we simply do not support it.

What currently happens is the following: If you configure with ```CMAKE_BUILD_TYPE=DebugRelease``` for a multi-configuration generator it happily creates 4 library targets: ```deal_II``` for DEBUG and RELEASE and ```deal_II.g``` for DEBUG and RELEASE. By the way how we set up the targets both flavors of the libraries, DEBUG and RELEASE, will be identical.

Therefore it is much better to not pretend that the debug/release switch in MSVC (or other generators) actually does something...